types/ws/index.d.ts

```diff
@@ -64,18 +66,22 @@ declare class WebSocket extends events.EventEmitter {
     // Events
     on(event: 'error', cb: (err: Error) => void): this;
     on(event: 'close', cb: (code: number, message: string) => void): this;
-    on(event: 'message', cb: (data: any, flags: { binary: boolean }) => void): this;
-    on(event: 'ping', cb: (data: any, flags: { binary: boolean }) => void): this;
-    on(event: 'pong', cb: (data: any, flags: { binary: boolean }) => void): this;
+    on(event: 'headers', cb: (headers: string[], request: http.IncomingMessage) => void): this;
```

Comment:
The `headers` argument is an object: https://github.com/websockets/ws/blob/master/doc/ws.md#event-headers-1.